### PR TITLE
Upgrade to Geoserver 2.15.0

### PR DESCRIPTION
**Pull request by groldan**
Upgrades `geoserver/webapp` to 2.15.0 as of #2605.
Note: I don't have write access to georchestra/geoserver to try this out you'll need to manually fetch the following branch:
```
cd geoserver/geoserver-submodule
git remote add groldan git@github.com:groldan/geoserver.git
git fetch groldan
git reset groldan/2.15.0-georchestra --hard
```

### CODE REVIEW

**Comment by RemiDesgrange** [comment]
Thank for this. As you are pointing in #2605 there are many stuff improved in geoserver 2.15 :
* java11 support that imply
  * better container support (jvm is aware of cgroups and namespaces)
  * availability of ZGC, which can lead to significant memory reduction (not sure, but we can try)
 * JAI-EXT by default
* ...

File: geoserver/webapp/pom.xml [hunk 1]
```diff
@@ -169,42 +169,6 @@
...
-      <artifactId>spring-tx</artifactId>
-      <version>${gs.spring.version}</version>
-    </dependency>
-    <!-- xml-apis is needed by gt-svg in v1.3.04, but the GS webapp upstream
-         uses v1.4.01 -->
-    <dependency>
-      <groupId>xml-apis</groupId>
-      <artifactId>xml-apis</artifactId>
-      <version>1.4.01</version>
-    </dependency>
```
Comment (pmauduit) [inline geoserver/webapp/pom.xml]: Why removing the dependencies ? Are they now implicit by using the spring boot parent pom ?

File: geoserver/lib/2.15.0/gt-s3-geotiff-21.0.jar.readme [hunk 2]
```diff
@@ -0,0 +1,8 @@
+This is gt-s3-getotiff-21-SNAPHOT meant to
+replace gt-s3-geotiff-21.0.jar in 
+geoserver/WEB-INF/lib, with fixes
+that didn't make it to the official 21.0 
+GeoTools release from Tue Feb 2019, whilst
+the patch is from Mon Mar 2019 at commit
+c73b886b1, and hence will be available in
+the next 21.1 GeoTools release.
```
Comment (pmauduit) [inline geoserver/lib/2.15.0/gt-s3-geotiff-21.0.jar.readme]: I'm not very fond of adding binaries to git, but It's fine. Do you confirm that we could remove them later on ? (once GS will have upgraded the GT version, and we will stick with the new one) ?

**Comment by fvanderbiest** [comment]
Hi Gabriel, what's the status of this PR ? Is it draft or ready ?

> I don't have write access to georchestra/geoserver

Weird, since you're a commiter ...

File: geoserver/lib/2.15.0/gt-s3-geotiff-21.0.jar.readme (hunk 2 above)
Comment (groldan) [reply geoserver/lib/2.15.0/gt-s3-geotiff-21.0.jar.readme]: yeah, we just need to upgrade to a version that comes with the patched s3-geotiff.jar. Which, as of today, should be 15.1, I'll give it a try without the extra binaries and update the PR

**Pull request merged**